src/jp/hidetobara/pasteltouch/PastelTouchActivity.java

```diff
@@ -0,0 +1,535 @@
+		if( scale > 1.0f ) scale = 1.0f;
+		m.postScale(scale, scale);
+    	return m;
+    }
+    
+    private void assignBitmap()
+    {
+    	if(_Bitmap == null || _Canvas == null) return;
+
+    	_Canvas.setBitmap(_Bitmap);	//Bitmapの適応
+		_Canvas.setActivity(this);	//activityの登録
+		_Canvas.reinitialize();		//初期化
+		_Canvas.setBitmapFit(); 	//表示をFitさせる
+    }
+    
+    public void registerFileToGallery(File file)
+    {
+    	if( file == null ) return;
+    	
+        String filepath = file.getPath();
+        String filename = file.getName();
+        String suffix   = Utility.getSuffix(file);
+        String mimetype = "";
+        
+        if ((suffix.equals("jpg")) || (suffix.equals("jpeg"))) {
+            mimetype = "image/jpeg";
+        }else if (suffix.equals("gif")) {
+            mimetype = "image/gif";
+        }else if (suffix.equals("png")) {
+            mimetype = "image/png";
+        }
+        
+        if (mimetype != "") {
+            ContentResolver contentresolver = this.getContentResolver();
+            ContentValues contentvalues = new ContentValues();
+            contentvalues.put(MediaStore.Images.Media.MIME_TYPE, mimetype);
+            contentvalues.put(MediaStore.Images.Media.DATA, filepath);
+            contentvalues.put(MediaStore.Images.Media.TITLE, filename);
+            contentvalues.put(MediaStore.Images.Media.DISPLAY_NAME, filename);
+            Uri uri = contentresolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentvalues);
+            dp(uri.getPath());
+        }else{
+        	dp("Fail: ギャラリーへの登録");
+        }
+    }
+
+    public void unregisterFileToGallery(String name)
+    {
+    	if( name.length()==0 ) return;
+    	
+    	String path = Utility.getRootPath() + "/" + name + ".png";
+		ContentResolver resolver = this.getContentResolver();
+		int affect = resolver.delete(
+				MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+				MediaStore.Images.Media.DATA + " = '" + path + "'", null);
+		if(affect == 0) dp("Fail: ギャラリーへの抹消");
+    }
+    
+    /*
+     * URI から path へ変換する
+     */
+    private String getUriToPath(Uri uri)
+    {
+		Cursor c = getContentResolver().query(uri, null, null, null, null);
+		if(c == null) return "";
+		c.moveToFirst();
+		return c.getString(c.getColumnIndex(MediaStore.MediaColumns.DATA));
+    }
+    
+    /*
+     * Pastel Touch 用の名前かどうか調べる
+     */
+    private String checkPastelTouchName(String path)
+    {
+    	String[] cells = path.split("/");
+    	if(cells.length < 2) return null;
+    	
+    	String appname = cells[cells.length-2];
+    	String name = cells[cells.length-1];
+    	if(!appname.equals(APPLICATION_NAME)) return null;
+    	
+    	int last = name.lastIndexOf(".");
+    	String head = name.substring(0, last);
+    	
+    	return head;
+    }
+        
+    private String getCurrentName()
+    {
+    	Date date = new Date(System.currentTimeMillis());
+    	SimpleDateFormat format = new SimpleDateFormat("yyyyMMddHHmmss");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.